package pylongps_protobuf_sql_converter;


enum test_enum
{
TEST_OFFICIAL = 1; //Base stations that are run by trusted entities (such as governments) and use authentication so that injecting bad data is difficult
TEST_REGISTERED_COMMUNITY = 2; //Base stations run by private individuals or organizations that have registered and gotten a signing key for their organization
TEST_COMMUNITY = 3; //Base stations run by anonymous individuals
} 

message protobuf_sql_converter_test_message
{
optional int64 optional_int64 = 10;
optional double optional_double = 20;
optional bytes optional_string = 30;
optional test_enum optional_enum = 40;

//These are left null to test that functionality
optional int64 null_optional_int64 = 11; 
optional double null_optional_double = 21;
optional bytes null_optional_string = 31;
optional test_enum null_optional_enum = 41;

required int64 required_int64 = 50;
required double required_double = 60;
required bytes required_string = 70;
required test_enum required_enum = 80;

repeated int64 repeated_int64 = 90;
repeated double repeated_double = 100;
repeated bytes repeated_string = 110;
repeated test_enum repeated_enum = 120;
}
